plugins/inventory/nb_inventory.py

```diff
@@ -1578,7 +1583,11 @@ def main(self):
                 # Device is part of a virtual chassis, but is not the master
                 continue
 
-            hostname = self.extract_name(host=host)
+            if self.ansible_inventory_primary_ip and self.extract_primary_ip(host=host):
+                hostname = self.extract_primary_ip(host=host)
```

Comment:
This is a little hacky. I don't know if I like setting a variable named `hostname` with a raw IP address in some cases.

Reply:
I agree with you. Unfortunately Ansible itself has IP addresses straight on the inventory in their documentation, so any var used for that could contain an hostname or an IP.

Reply:
Yes but my nit is that in most of the Ansible internals, they use `host` - which could either be an IP or a DNS name. You are assigning an IP address to a variable called `hostname` which, in my opinion, breaks the meaning of the variable.